### PR TITLE
update Dev Setup instructions

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
Updated the dev setup guide.
- made improvements to the overall flow of the guide
- split out lengthy fixes for Mac OS specific stuff into its own file
- added instructions for how to update data in elasticsearch indices manually on macs since `run_ptop` still fails
- also included is a better way of managing Java SDKs on mac with `sdkman` instead of `jenv`
- now recommend `pyenv` instead of `virtualenvwrapper` (for reasons you will see in the setup instructions!)

### Safety story
Only `.md` files have been touched.

### Automated test coverage
No

### QA Plan
No

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AddisonDunn — APPROVED** [review]
I'm not familiar with sdkman or pyenv so I skipped over some of that but I love the separate Mac stuff!

File: DEV_SETUP_MAC.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+To fix:
+```sh
+cd /usr/local/lib/
+ln -s /opt/homebrew/Cellar/libmagic/5.41/lib/libmagic.dylib libmagic.dylib
+```
+
+## Refreshing data in `elasticsearch` manually (alternative to `run_ptop`)
+
+To refresh specific indices in elasticsearch you can do the following...
```
[the remark this answers is not in the document]
Comment (biyeun) [reply DEV_SETUP_MAC.md:184]: thanks!

File: DEV_SETUP_MAC.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+```
+
+Force a re-index of `forms` and `cases`:
+```sh
+./manage.py ptop_reindexer_v2 sql-case --reset
+./manage.py ptop_reindexer_v2 sql-form --reset
+```
+
+For other indices (pick and choose):
+```sh
```
[the remark this answers is not in the document]
Comment (biyeun) [reply DEV_SETUP_MAC.md:201]: yeah that's better :)

[the remark this answers is not in the document]
Comment (biyeun) [reply DEV_SETUP_MAC.md:201]: (although i like it for the quick copy paste value...but my private notes are for that)

**Pull request merged**